### PR TITLE
[FL-2649] Drop Docker in CI/CD

### PR DESCRIPTION
**Pull request by drunkbatya**
# What's new

- Excluding Docker from CI/CD
- Add [Black python formatter](https://github.com/psf/black) to our toolchain
- Add FBT_TOOLCHAIN_PATH env variable to prevent toolchain redownloading on github-runners
- Add Rosseta2 check on MacOS under Apple Silicon
- Add x86_64 arch check on Linux
- Add './fbt cli' target, to easy access Flipper CLI with PySerial miniterm

# Verification 

- All GitHub's workflows are working
- All ./fbt lint targets like 'lint_py' and 'format_py' also should work
- Warn message with Rosetta installing instructions on ./fbt under Apple Silicon MacOS with no Rosetta (exit code 1)
- Warn message about wrong architecture on ./fbt under non-x86_64 Linux
- './fbt cli' should detect connected flipper and open serial connection to Flipper CLI in current shell

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: site_scons/_black_version.py [hunk 1]
```diff
@@ -0,0 +1 @@
+version = "22.6.0"
```
Comment (hedger) [inline site_scons/_black_version.py]: no longer needed

File: site_scons/site_init.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import atexit
 
 sys.path.insert(0, os.path.join(os.getcwd(), "scripts"))
+sys.path.insert(0, os.path.join(os.getcwd(), "lib/black/src"))
```
Comment (hedger) [inline site_scons/site_init.py]: no longer needed

File: scripts/toolchain/fbtenv.sh [hunk 3]
```diff
@@ -1,54 +1,195 @@
...
+    elif [ "$(cat "$TOOLCHAIN_ARCH_DIR/VERSION")" -ne "$FBT_TOOLCHAIN_VERSION" ]; then
+        fbtenv_download_toolchain || return 1;
+    fi
+    return 0;
+}
+
+fbtenv_show_usage()
+{
+    echo "fbtenv.sh - sets eviroment variables for Flipper Build Tool (FBT) and downloads toolchain";
+    echo "Warning! Not recomended to run this script mannualy! Use ./fbt !";
```
Comment (hedger) [inline scripts/toolchain/fbtenv.sh]: typo: manually

File: scripts/lint_py.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
```
Comment (hedger) [inline scripts/lint_py.py]: whole file: no longer needed

File: scripts/toolchain/fbtenv.sh [hunk 5]
```diff
@@ -1,54 +1,195 @@
...
+    echo "Usage:";
+    printf "\tFBT_TOOLCHAIN_VERSION=.. FBT_TOOLCHAIN_PATH=.. SCRIPT_PATH=.. source fbtenv.sh\n";
+    echo "Example:";
+    printf '\tFBT_TOOLCHAIN_VERSION=10 SCRIPT_PATH=$(pwd) FBT_TOOLCHAIN_PATH=/tmp/ source scripts/toolchain/fbtenv.sh\n'
```
Comment (hedger) [inline scripts/toolchain/fbtenv.sh]: can we have a simpler way to run fbtenv? use defaults when sourcing from repo root?

**Review by hedger — CHANGES_REQUESTED** [review]

**Review by hedger — APPROVED** [review]

File: .github/workflows/check_submodules.yml [hunk 6]
```diff
@@ -1,17 +0,0 @@
-name: 'Check submodules'
```
Comment (hedger) [inline .github/workflows/check_submodules.yml:1]: This check is still needed

File: scripts/toolchain/fbtenv.sh [hunk 7]
```diff
@@ -1,54 +1,211 @@
...
+    echo "Running this script manually is wrong, please source it";
+    echo "Example:";
+    printf "\tsource scripts/toolchain/fbtenv.sh\n";
+    return 1;
+}
+
+fbtenv_check_script_path()
+{
+    if [ ! -x "$SCRIPT_PATH/fbt" ]; then
+        echo "Please source this script being into flipperzero-firmware root directory, or specify 'SCRIPT_PATH' manualy";
```
Comment (hedger) [inline scripts/toolchain/fbtenv.sh]: >manualy 
typo

**Review by hedger — CHANGES_REQUESTED** [review]

File: scripts/toolchain/fbtenv.sh [hunk 8]
```diff
@@ -19,7 +19,7 @@ fbtenv_check_sourced()
     if [ "$(basename $0)" = "fbt" ]; then
         return 0;
     fi
-    echo "Running this script manually is wrong, please source it";
```
Comment (anna-oake) [inline scripts/toolchain/fbtenv.sh]: this was correct

Comment (drunkbatya) [reply scripts/toolchain/fbtenv.sh]: Thanks, fixed

**Review by hedger — DISMISSED** [review]

**Review by hedger — APPROVED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**